music21/musicxml/xmlToM21.py

```diff
@@ -2850,6 +2865,15 @@ def xmlToNote(self, mxNote: ET.Element) -> None:
         self.offsetMeasureNote += offsetIncrement
         self.endedWithForwardTag = None
 
+        # reset offset for figures. This is needed to put in
+        # multiple FiguredBassIndications at one note at the right offset.
+        # Musicxml puts <figured-bass> tags immediately before a <note> tag,
+        # which means that we have to reset a given offset duration of some
+        # <figured-bass> tags after inserting the coressponding note and
+        # before going to a new note.
+        self.lastFigureDuration = 0
```

Comment:
I would appreciate if you could share more of your findings around duration/offset/tstamp. It's not yet clear to me why we can't just treat the `<figured-bass>` tags like all the other musicxml tags and just insert them into the measure as we go, like almost all the other musicxml methods in this class. It's true, if there are more than one at the same time point, we need to adjust the last one instead of creating a new one, but that's just a couple lines of code (e.g. `if self.stream.getElementsByOffset(self.offsetInMeasure, classList=[Figured...])`